streams/src/main/java/org/apache/kafka/streams/processor/internals/RecordCollectorImpl.java

```diff
@@ -125,6 +127,7 @@ private boolean productionExceptionIsFatal(final Exception exception) {
     ) {
         String errorLogMessage = LOG_MESSAGE;
         String errorMessage = EXCEPTION_MESSAGE;
+        // There is no documented API for detecting errors that could be solved by retrying, so we do the best we can
```

Comment:
What does this comment mean?

Reply:
`RetriableException` is not documented in any of the clients javadoc. We are relying on an implementation detail here, but we have no better alternative.